### DISCUSSION TITLE
Outline a text when using convert and draw

### DISCUSSION THREAD

**Discussion by naromwerd** [ImageMagick/ImageMagick#4299] [Help]
the command I am using is:
magick convert -font Arial -pointsize 20 -fill yellow -draw "text 875,570 'Hello World'" canvas.tif canvas_data.tif
Is there a way to draw a border around the text "hello world"? I cant find an option.

Thanks

**Comment by fmw42** [comment]
Post canvas.tif. Also explain where you want the border?  Is that about the whole outside of the text or for each character? If the former, is the border rectangular or follow the shape of the characters in the words? Can you post an example of what you want?  Perhaps

magick lena.png \( -background none -font Arial -pointsize 20 -fill yellow label:"Hello World" -compose copy -bordercolor blue -border 2 \) -geometry +50+100 -compose over -composite result.png

![result](https://user-images.githubusercontent.com/13458218/135554079-4fe6069b-2307-4aeb-8ec3-69c7698832cd.png)

The above is Unix syntax. For Windows remove the \\ from \\(  and from \\)

The above used label: to convert the text to an image with transparent background. Then adds a blue border. Then that result is composited over the background lena image.

Note: For ImageMagick 7, use magick, not magick convert.

**Reply by naromwerd** [reply]
Sorry, Thoought I had enough detail in my question, maybe this is better!
basic commands i am using are as follow:
    'magick convert -size 1280x1024 xc:none canvas.tif
    'magick convert -font Arial -pointsize 20 -fill yellow -draw "text 472,612 'Text1'" -draw "text 789,704 'Text2'" canvas.bmp canvas_data.bmp
    'magick composite -gravity center canvas_data.bmp inputfile.bmp outputfile.bmp
inputfile.bmp is what I intend to overlay with the transparent canvas_data.bmp file.
So how to put a border just around Text2?
Thanks

**Comment by GeeMack** [comment]
Using "magick convert" will make the program behave as if it's ImageMagick v6. Unless there's a particular reason for doing that, it is generally recommended to just use "magick".

Also, it is best practice to read in the image before doing any operations, and with IMv7 being pickier about syntax, it's probably a good habit to get into.

    magick inputimage,png ... other operations ... outputimge.png

To make a thin black outline around the text, add "`-stroke black -strokewidth 2`" ahead of your "-draw" operation.

**Reply by naromwerd** [reply]
Thanks for that. I am just starting to use this and had not read that yet!
